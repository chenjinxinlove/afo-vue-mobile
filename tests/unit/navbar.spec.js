import Navbar from '../../src/navbar/index'
import Vue from 'vue'
import Demo from '../../src/Navbar/demo'
import {shallowMount, mount} from '@vue/test-utils'

describe('Navbar.vue', () => {
  // 测试是否可以被使用
  it('use', () => {
    Vue.use(Navbar)
    expect(Vue.component(Navbar.name))
      .toHaveProperty('name', 'VueComponent')
  })
  // 是否能正确渲染
  it('props', () => {
    const wrapper = shallowMount(Navbar)
    expect(wrapper.classes()).toEqual(['afo-navbar'])
  })
  // 测试props
  it('props', () => {
    const wrapper = shallowMount(Navbar, {
      propsData: {
        title: 'title',
        fixed: true
      },
      slots: {
        left: '<button>left</button>',
        right: '<button>right</button>'
      }
    })
    expect(wrapper.classes()).toEqual(['afo-navbar', 'is-fixed'])
  })
  // demo测试快照
  it('demo snapshot', () => {
    expect(mount(Demo).html()).toMatchSnapshot()
  })
})
