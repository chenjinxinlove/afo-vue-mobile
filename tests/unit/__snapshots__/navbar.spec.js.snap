// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar.vue demo snapshot 1`] = `
<div>
  <header class="afo-navbar is-fixed">
    <div class="afo-navbar-button is-left">
      <button type="button" class="afo-button afo-button--default afo-button--normal">
        <i class="afo-icon-back"></i>
        <!---->返回</button>
    </div>
    <h1 class="afo-navbar-title">我是fixed标题</h1>
    <div class="afo-navbar-button is-right">
      <button type="button" class="afo-button afo-button--default afo-button--normal">
        <i class="afo-icon-more"></i>
        <!---->
      </button>
    </div>
  </header>
  <header class="afo-navbar">
    <div class="afo-navbar-button is-left">
      <button type="button" class="afo-button afo-button--default afo-button--normal">
        <i class="afo-icon-back"></i>
        <!---->返回</button>
    </div>
    <h1 class="afo-navbar-title">我是标题</h1>
    <div class="afo-navbar-button is-right">
      <button type="button" class="afo-button afo-button--default afo-button--normal">
        <i class="afo-icon-more"></i>
        <!---->
      </button>
    </div>
  </header>
</div>
`;
